src/pages/NewChatPage.js

```diff
@@ -129,6 +132,26 @@ class NewChatPage extends Component {
         return sections;
     }
 
+    /**
+     * This will find an existing chat, or create a new one if none exists, for the given user or set of users. It will then navigate to this chat.
+     *
+     * @param {Object} option option.
+     */
+    getOrCreateChatReport(option) {
+        const userLogins = _.pluck(option, 'login');
+        const formattedUserLogins = _.map(userLogins, login => OptionsListUtils.addSMSDomainIfPhoneNumber(login).toLowerCase());
+        let newChat = {};
+        let newPersonalDetail = {};
+        const chat = ReportUtils.getChatByParticipants(formattedUserLogins);
+        if (!chat) {
+            newChat = ReportUtils.buildOptimisticChatReport(formattedUserLogins);
+            newPersonalDetail = PersonalDetails.buildNewPersonalDetails(option);
+        }
+        const reportID = chat ? chat.reportID : newChat.reportID;
+        Report.openReport(reportID, newChat.participants, newChat, newPersonalDetail);
```

Comment:
I am not sure why we have `Report.openReport` and `Navigation` call, and we also have `Report.navigateToAndOpenReport` in `createChat` method.

Do we need this twice?